media/CircleEditor/view-sidebar.js

```diff
@@ -1456,30 +1439,269 @@ sidebar.ArgumentView = class {
+            this.changeBufferType('string', modified, shape);
+        } else {
+            this.compareData(
+                modifiedArr, originalArr, this._editObject._arguments._initializer._data,
+                currentType.toLowerCase());
+        }
+    }
 
-    /**
-     * TODO Implement editBuffer()
-     * Buffer edit function
-     */
+    /* buffer type modified - data change NOT detected */
+    changeBufferType(newType, modified, shape) {
+        // original =  textarea.value
+        const modifiedArr = this.parseBufData(modified, shape, newType);
 
-    /**
-     * TODO Implement changeBufferType()
-     * Buffer tyep change function
-     */
+        if (modifiedArr.length === 0) {
+            return;
+        }
+
+        const buffer = new ArrayBuffer(8);
+        const view = new DataView(buffer);
+
+        let newArray = [];
+
+        if (newType.startsWith('float')) {
+            const bits = Number(newType.slice(-2));
+
+            for (let i = 0; i < modifiedArr.length; i++) {
+                const data = parseFloat(modifiedArr[i]);
+
+                if (isNaN(data)) {
+                    return false;
+                } else if (bits === 16) {
+                    view.setFloat16(0, data, true);
+                } else if (bits === 32) {
+                    view.setFloat32(0, data, true);
+                } else if (bits === 64) {
+                    view.setFloat64(0, data, true);
+                }
+
+                for (let j = 0; j < bits / 8; j++) {
+                    newArray.push(view.getUint8(j));
+                }
+            }
+        } else if (newType.startsWith('int')) {
+            let bits = Number(newType.slice(-2));
+
+            for (let i = 0; i < modifiedArr.length; i++) {
+                const data = parseInt(modifiedArr[i]);
+
+                if (isNaN(data)) {
+                    return false;
+                } else if (bits === 16) {
+                    view.setInt16(0, data, true);
+                } else if (bits === 32) {
+                    view.setInt32(0, data, true);
+                } else if (bits === 64) {
+                    view.setBigInt64(0, BigInt(data), true);
+                } else {
+                    bits = Number(newType.slice(-1));
+
+                    if (bits === 8) {
+                        view.setInt8(0, data, true);
+                    }
+                }
+                for (let j = 0; j < bits / 8; j++) {
+                    newArray.push(view.getUint8(j));
+                }
+            }
+        } else if (newType.startsWith('uint')) {
+            let bits = Number(newType.slice(-2));
+
+            for (let i = 0; i < modifiedArr.length; i++) {
+                const data = parseInt(modifiedArr[i]);
+
+                if (data < 0) {
+                    return false;
+                }
+
+                if (isNaN(data)) {
+                    return false;
+                } else if (bits === 32) {  // uint32
```

Comment:
16 is missed maybe?

Reply:
I checked `Samsung/ONE/circle_schema.fbs` but they did not have uint16, so I omitted it on purpose. Should I add it?:)

Reply:
Oh... I am sorry, never mind my comment :(